File: ILT.IHR.DTO/UserRole.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public class UserRole : AbstractDataObject
    {
        public int UserRoleID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        public int UserID { get; set; }
        public int RoleID { get; set; }
        public string RoleShort { get; set; }
        public string RoleName { get; set; }
        public bool IsDefault { get; set; }
        public bool IsSelected { get; set; }
    }
}

File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/UserRole.ts

export class IUserRole {
    UserRoleID: number;
    UserID?: number;
    RoleID: number;
    RoleShort: string;
    RoleName: string;
    IsDefault: boolean;
    IsSelected: boolean;
    CreatedDate: Date;
    CreatedBy: string;
    ModifiedDate: Date;
    ModifiedBy: string;
}
export class IUserRoleDisplay {
    userRoleID: number;
    userID?: number;
    roleID: number;
    roleShort: string;
    roleName: string;
    isDefault: boolean;
    isSelected: boolean;
    createdDate: Date;
    createdBy: string;
    modifiedDate: Date;
    modifiedBy: string;
}
